SOURCES/etc/xapi.d/plugins/linstor

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import re
+
+import XenAPI
+import XenAPIPlugin
+
+from xcpngutils import configure_logging, run_command, error_wrapped, strtobool
+
+# ------------------------------------------------------------------------------
+
+LVM_PLUGIN = 'lvm'
+THIN_POOL = 'thin_pool'
+BACKING_DISK_RE = re.compile('^/dev/([^/]+)/(?:[^/]+)$')
+
+# ------------------------------------------------------------------------------
+
+def _create_sr(name, description, disks, volume_group, redundancy, thin, force):
+    disks = json.loads(disks)
+    disk_hostnames = disks.keys()
+
+    try:
+        session = XenAPI.xapi_local()
+        session.xenapi.login_with_password('root', '', '', 'linstor-installer')
```

Comment:
A logged session is already given to all plugins methods, you can give it to this method as an args